Pi_Code/Controller_Operations.py

```diff
@@ -0,0 +1,113 @@
+		# where we click the "start" button on the joystick to begin the operation of the sub
+        self.read_stick = True
+        self.autosub = True
+
+        # Get the properties of the joystick
+        # Not necessary currently, may be useful later
+        print("Found joystick: " + self.controller.get_name())
+        buttons = self.controller.get_numbuttons()
+        axes = self.controller.get_numaxes()
+        balls = self.controller.get_numballs()
+        hats = self.controller.get_numhats()
+        p = Process(target = self.getStickPosition, args = (dataFile, connection, SaveSensorData))
+        p.start()
+
+    def getStickPosition(self, connection, dataFile, SaveSensorData):
+
+        while self.idle:
+            # Start button code. Hold in idle until button 6 is pressed, and then send the response to the main loop
+            pygame.event.pump()
+            start = self.controller.get_button(6)
+            if start:
+                connection.send("auto")
+                self.idle = False
+
+        while self.autosub:
+            # If we press button 5, we will switch to manual controls.
+			# Otherwise, it will just keep looping in here.
+            pygame.event.pump()
+            manual = self.controller.get_button(5)
+            if manual:
+                connection.send("manual")
+                self.autosub = False
+                self.read_stick = True
+
+        while self.read_stick:
+            # Update states
+            pygame.event.pump()
+
+            # Hardcoded for now, will fix later.
+            threshhold = .3
+
+            roll = self.controller.get_axis(0)
+            pitch = self.controller.get_axis(1)
+            throttle = self.controller.get_axis(2)
+            # yaw = controller.get_axis(3) (Not used at the moment)
+            trigger = self.controller.get_button(0)
+            stop = self.controller.get_button(6)
+            kill = self.controller.get_button(1)
+            auto_revert = self.controller.get_button(5)
+
+            # Now that we have pitch/roll/throttle/yaw values, we can start using them
+            # Send a tuple (x,y) where x and y are the speeds for the left and right motors respectively
+
+            if kill:
+                # Shut down both the motors
+                connection.send((0, 0))
+
+            elif trigger:
+                # Set the speed to throttle and sent it to both motors
+                connection.send((throttle, throttle))
+
+            elif abs(pitch) > threshhold:
+                # |Max pitch| is ~.6, and the max angle we should have for our fins is 20 deg (from Stubley)
+                # Divide the pitch value by 3, so our min dive angle for our fins will be 10 deg, and max will be 20
+                angle = pitch/3
+                connection.send(angle)
+
+            elif (not trigger) and (abs(roll) > threshhold):
```

Comment:
why is the roll check different from the pitch one?

Reply:
Both motors can be running while we adjust the servo angle, but one motor must be forced off while we are turning. The `(not trigger)` means we can only turn when we are not commanding both motors to go forward.